src/Raven.Server/Documents/Indexes/MapReduce/ReduceKeyProcessor.cs

```diff
@@ -137,6 +137,58 @@ public void Process(object value)
                 return;
             }
 
+            if (value is int)
+            {
+                var i = (int)value;
+
+                switch (_mode)
+                {
+                    case Mode.SingleValue:
```

Comment:
Why? We can cast it to long and not do any additional work.
Same for double and long, as long as it is a single value.